src/src/beneficiaries/beneficiaries.service.ts

```diff
@@ -902,6 +930,105 @@ export class BeneficiariesService {
 		// return this.hasuraService.delete(this.table, { id: +id });
 	}
 
+	public async deactivateDuplicateBeneficiaries(AadhaarNo: string, exceptId: number, createdBy: number) {
+		// Store previous data before update
+		const getQuery = `
+			query MyQuery {
+				users (
+					where: {
+						aadhar_no: {_eq: "${AadhaarNo}"}
+					}
+				) {
+					id
+					aadhar_no
+					is_duplicate
+					is_deactivated
+					duplicate_reason
+				}
+			}
+		`;
+
+		const preUpdateData = (await this.hasuraServiceFromServices.getData({ query: getQuery })).data?.users;
+		const preUpdateDataObj = {};
+		preUpdateData.forEach(userData => preUpdateDataObj[userData.id] = userData);
+		const query = `
+			mutation MyMutation {
+				update_users_many (
+					updates: [
+						{
+							where: {
+								aadhar_no: {_eq: "${AadhaarNo}"},
+								id: {_neq: ${exceptId}}
+							},
+							_set: {
+								is_deactivated: true
+							}
+						},
+						{
+							where: {
+								id: {_eq: ${exceptId}},
+							},
+							_set: {
+								is_deactivated: false
+							}
+						}
+					]
+				) {
+					returning {
+						id
+						aadhar_no
+						is_duplicate
+						is_deactivated
+						duplicate_reason
+					}
+				}
+			}
+		`;
+
+		const updateResult = (await this.hasuraServiceFromServices.getData({ query }))?.data?.update_users_many;
+
+		// Add audit logs of is_duplicate flag
+		await Promise.allSettled(
+			updateResult.map(updatedData => Promise.allSettled(updatedData.returning.map(updatedUserObj =>
+				this.userService.addAuditLog(
+					updatedUserObj.id,
+					createdBy,
+					'users.is_duplicate',
+					updatedUserObj.id,
+					{
+						is_duplicate: preUpdateDataObj[updatedUserObj.id].is_duplicate,
+						duplicate_reason: preUpdateDataObj[updatedUserObj.id].duplicate_reason,
+					},
+					{
+						is_duplicate: updatedUserObj.is_duplicate,
+						duplicate_reason: updatedUserObj.duplicate_reason,
+					},
+					['is_duplicate', 'duplicate_reason']
+				)
+			)))
+		);
+
+		// Add audit logs of is_deactivated flag
+		await Promise.allSettled(
+			updateResult.map(updatedData => Promise.allSettled(updatedData.returning.map(updatedUserObj =>
+				this.userService.addAuditLog(
+					updatedUserObj.id,
```

Comment:
can we set this data into above block itself

it is OK to add more than one fields into changeset in one go

Reply:
Done

Reply:
As we're storing two different context like `users.is_duplicate` and `users.is_deactivated`. This change-set is related to `users.is_deactivated`. That's why I've kept it separate from above set of `users.is_duplicate`.

Reply:
Done